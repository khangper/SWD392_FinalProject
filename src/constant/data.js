export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
