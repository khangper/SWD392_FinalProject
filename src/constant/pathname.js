export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
