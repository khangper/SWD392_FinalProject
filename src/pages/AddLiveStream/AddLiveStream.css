/* AddLiveStream.css */

/* Container */
.container-fluid {
    display: grid;
    grid-template-columns: 1fr;
    gap: 20px;
    justify-content: center;
    align-items: center; /* Center vertically */
    min-height: 100vh; /* Ensure container spans full viewport height */
    user-select: none;
}

  /* Row */
  .row {
    display: grid;
    grid-template-columns: 1fr;
    justify-content: center;
  }
  
  /* Columns */
  .col-lg-8, .col-md-10 {
    grid-column: span 1;
    justify-self: center;
  }
  


.section3125 {
    float: left;
    width: 100%;
}


.nav-tabs {
    display: flex;
    justify-content: center;
    list-style-type: none; /* Remove default list styling */
    padding: 0; /* Remove default padding */
  }
  
  .nav-item {
    margin: 0 10px; /* Add spacing between tabs */
  }
  
/* AddLiveStream.css */

.nav-link {
    padding: 10px 20px;
    cursor: pointer;
    color: white; /* Màu chữ mặc định */
    background-color: #dc3545; /* Màu nền khi chưa được chọn */
    border-radius: 5px; /* Đường viền bo tròn */
  }
  
  .nav-link.active {
    background-color: black; /* Màu nền khi được chọn */
  }
  

  .add_stream_content {
    background: #fff;
    margin-top: 30px;
    padding: 30px;
    border-radius: 10px;
    border: 1px solid #efefef;
    margin-bottom: 20px;
    width: 100%;
    max-width: 846px;
    margin-left: auto;
    margin-right: 254px;
    height: auto;
}



.strm_title {
    font-size: 18px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #333;
    line-height: 24px;
    margin-bottom: 0;
}

.strm_title1 {
    font-size: 18px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    text-align: left;
    color: #333;
    line-height: 24px;
    margin-bottom: 0;
}

.sf475 {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    text-align: center;
    margin-top: 5px;
    line-height: 24px;
}

.sf475 a {
    font-weight: 500;
    color: #333;
}

.sf475 a:hover {
    color: #ed2a26 !important;
}

.stm_key {
    font-size: 18px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    margin-top: 30px;
    color: #333;
    line-height: 24px;
}

.group-form {
    margin-top: 27px;
	position: relative;
}

.group-form label {
	font-size: 14px;
	font-weight: 500;
	font-family: 'Roboto', sans-serif;
    margin-bottom: 10px;
    color: #333;
    line-height: 24px;
}

._dlor1 {
    border: 1px solid #efefef;
    background: #f7f7f7;
    height: 40px;
    width: calc(100% - 120px); /* Đặt giá trị mới cho chiều rộng */
    padding: 0 20px; /* Điều chỉnh padding để không che khuất nội dung */
    border-radius: 3px;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    font-size: 13px;
    box-sizing: border-box; /* Đảm bảo tính chính xác của kích thước */
    white-space: nowrap; /* Ngăn chặn quá trình quy xuống dòng */
    overflow: hidden; /* Ngăn chặn hiển thị nội dung ra ngoài */
    text-overflow: ellipsis; /* Hiển thị dấu ba chấm khi nội dung bị cắt bớt */
}



._145d1 {
    height: 40px;
    padding: 0 20px;
    background: #ed2a26;
    color: #fff;
    border: 0;
	margin-top: 30px;
    border-radius: 3px;
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
}

._145d1:hover {
    background: #333;
    color: #fff;
}

._145d1 i {
    font-size: 120%;
	margin-right: 3px;
}

/* .ui.toggle._1457s2.checkbox input:checked~.box:before, .ui.toggle._1457s2.checkbox input:checked~label:before {
    background-color: #ed2a26 !important;
}

.ui.toggle._1457s2.checkbox input:focus:checked~.box:before, .ui.toggle._1457s2.checkbox input:focus:checked~label:before {
    background-color: #ed2a26 !important;
}

.ui.toggle.checkbox .box:before, .ui.toggle.checkbox label:before {
    display: block;
    position: absolute;
    content: '';
    z-index: 1;
    -webkit-transform: none;
    transform: none;
    border: none;
    top: 0;
    background: rgba(0, 0, 0, .05);
    -webkit-box-shadow: none;
    box-shadow: none;
    width: 3.5rem;
    height: 1.5rem;
    border-radius: 500rem;
} */
.ui.toggle.checkbox {
    display: flex;
    align-items: center;
    margin: 10px 0;
}

.ui.toggle.checkbox input[type="checkbox"] {
    display: none; /* Ẩn checkbox để sử dụng custom toggle */
}

.ui.toggle.checkbox label {
    cursor: pointer;
    display: inline-block;
    position: relative;
    padding-left: 50px;
    line-height: 24px;
    font-size: 14px;
}

.ui.toggle.checkbox label:before,
.ui.toggle.checkbox label:after {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 40px;
    height: 24px;
    transition: background-color 0.4s, transform 0.4s;
}

.ui.toggle.checkbox label:before {
    background-color: #ccc;
    border-radius: 12px;
}

.ui.toggle.checkbox label:after {
    background-color: #fff;
    width: 20px;
    height: 20px;
    border-radius: 50%;
    top: 2px;
    left: 2px;
}

/* Xử lý khi checkbox được chọn */
.ui.toggle.checkbox input[type="checkbox"]:checked + label:before {
    background-color: red;
}

.ui.toggle.checkbox input[type="checkbox"]:checked + label:after {
    transform: translateX(16px);
}

/* Xử lý khi checkbox chưa được chọn */
.ui.toggle.checkbox input[type="checkbox"]:not(:checked) + label:before {
    background-color: #ccc;
}

.ui.toggle.checkbox input[type="checkbox"]:not(:checked) + label:after {
    transform: translateX(0);
}

/* Xử lý khi checkbox được nhấn */
.ui.toggle.checkbox input[type="checkbox"]:active + label:before {
    background-color: blue;
}




._145d1 {
    height: 40px;
    padding: 0 20px;
    background: #ed2a26;
    color: #fff;
    border: 0;
	margin-top: 30px;
    border-radius: 3px;
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
}

._145d1:hover {
    background: #333;
    color: #fff;
}

._145d1 i {
    font-size: 120%;
	margin-right: 3px;
}

/* AddLiveStream.css */

.red-button {
    background-color: #dc3545; /* Màu nền đỏ */
    color: white; /* Màu chữ trắng */
    border: none; /* Loại bỏ đường viền */
    padding: 8px 12px; /* Điều chỉnh padding */
    border-radius: 5px; /* Đường viền bo tròn */
    margin-left: 10px; /* Khoảng cách với input */
  }
  .input-group {
    display: flex; /* Sử dụng flexbox để căn chỉnh các phần tử trong cùng một hàng */
    align-items: center; /* Căn chỉnh các phần tử theo chiều dọc */
  }
  

  
  ._6tf7s {
    background-color: #dc3545; /* Màu nền đỏ */
    color: white; /* Màu chữ trắng */
    border: none; /* Loại bỏ đường viền */
    padding: 10px 15px; /* Điều chỉnh padding */
    border-radius: 4px; /* Đường viền bo tròn */
    cursor: pointer; /* Con trỏ chuột thành dấu nhấp */
  }
  

  
  /* Thiết lập hover cho nút "Copy" */
  ._6tf7s:hover {
    background-color: #c82333; /* Màu nền đỏ sáng khi hover */
  }