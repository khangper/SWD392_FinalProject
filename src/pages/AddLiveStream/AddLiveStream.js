import React, { useState } from "react";
import "./AddLiveStream.css";

export default function AddLiveStream() {
  const [activeTab, setActiveTab] = useState("add-streaming");
  // Khởi tạo useState cho các checkbox ban đầu
  const [isChecked1, setIsChecked1] = useState(true);
  const [isChecked2, setIsChecked2] = useState(false);
  const [isChecked3, setIsChecked3] = useState(false);
  const [isChecked4, setIsChecked4] = useState(true);
  const [isChecked5, setIsChecked5] = useState(true);
  const [isChecked6, setIsChecked6] = useState(false);
  const [isChecked7, setIsChecked7] = useState(false);
  const handleTabChange = (tab) => {
    setActiveTab(tab);
  };

  return (
    <div className="sa4d25">
      <div className="Live-container-fluid">
        <div className="row justify-content-md-center">
          <div className="col-lg-8 col-md-10">
            <div className="section3125 stream_tabs">
              <ul
                className="nav nav-tabs slive_tabs justify-content-center"
                id="myTab"
                role="tablist"
              >
                <li className="nav-item">
                  <div
                    className={`nav-link ${
                      activeTab === "add-streaming" ? "active" : ""
                    }`}
                    onClick={() => handleTabChange("add-streaming")}
                    role="tab"
                    aria-controls="add-streaming"
                    aria-selected={activeTab === "add-streaming"}
                  >
                    Add Streaming
                  </div>
                </li>
                <li className="nav-item">
                  <div
                    className={`nav-link ${
                      activeTab === "setting" ? "active" : ""
                    }`}
                    onClick={() => handleTabChange("setting")}
                    role="tab"
                    aria-controls="setting"
                    aria-selected={activeTab === "setting"}
                  >
                    Setting
                  </div>
                </li>
              </ul>
              <div className="tab-content" id="myTabContent">
                {activeTab === "add-streaming" && (
                  <div
                    className="tab-pane fade show active"
                    id="add-streaming"
                    role="tabpanel"
                  >
                    <div className="add_stream_content">
                      <h4 className="strm_title">
                        Connect your live stream to the Live API
                      </h4>
                      <div className="sf475">
                        Use live-streaming software or a hardware encoder.{" "}
                        <a href="#">Learn More</a>
                      </div>
                      <div className="stm_key">
                        Preview your broadcast with a stream key or paired
                        encoder.
                      </div>
                      <div className="live_form">
                        <div className="group-form">
                          <label>Server URL*</label>
                          <div className="input-group">
                            <input
                              className="_dlor1"
                              type="text"
                              data-testid="server_url"
                              placeholder="Stream URL"
                              value="rtmps://live-api-s.edututs+.com:443/rtmp/"
                            />
                            <button
                              className="_6tf7s red-button"
                              type="button"
                              value="1"
                            >
                              Copy
                            </button>
                          </div>
                        </div>
                        <div className="group-form">
                          <label>Persistent stream key*</label>
                          <div className="input-group">
                            <input
                              className="_dlor1"
                              type="text"
                              data-testid="server_url"
                              placeholder="Stream URL"
                              value="592030151361629?s_bl=1&s_ps=1&s_sw=065&s_vt=api-s&a=AbzB7xYk7XdnKpBf"
                            />
                            <button
                              className="_6tf7s red-button"
                              type="button"
                              value="1"
                            >
                              Copy
                            </button>
                          </div>
                        </div>
                        <button className="_145d1">
                          <i className="uil uil-video"></i>Go Live
                        </button>
                      </div>
                    </div>
                  </div>
                )}
                {activeTab === "setting" && (
                  <div
                    className="tab-pane fade show active"
                    id="setting"
                    role="tabpanel"
                  >
                    <div className="add_stream_content">
                      <h4 className="strm_title1">Live Streaming Setting</h4>
                      <div className="ui toggle checkbox _1457s2">
                        <input
                          type="checkbox"
                          name="stream_ss1"
                          id="stream_ss1"
                          checked={isChecked1}
                          onChange={() => setIsChecked1(!isChecked1)}
                        />
                        <label htmlFor="stream_ss1">
                          Publish this as a continuous live video
                        </label>
                      </div>

                      <div className="ui toggle checkbox _1457s2">
                        <input
                          type="checkbox"
                          name="stream_ss2"
                          id="stream_ss2"
                          checked={isChecked2}
                          onChange={() => setIsChecked2(!isChecked2)}
                        />
                        <label htmlFor="stream_ss2">Allow embedding</label>
                      </div>

                      <div className="ui toggle checkbox _1457s2">
                        <input
                          type="checkbox"
                          name="stream_ss3"
                          id="stream_ss3"
                          checked={isChecked3}
                          onChange={() => setIsChecked3(!isChecked3)}
                        />
                        <label htmlFor="stream_ss3">
                          Unpublish after live video ends
                        </label>
                      </div>

                      <div className="ui toggle checkbox _1457s2">
                        <input
                          type="checkbox"
                          name="stream_ss4"
                          id="stream_ss4"
                          checked={isChecked4}
                          onChange={() => setIsChecked4(!isChecked4)}
                        />
                        <label htmlFor="stream_ss4">
                          Allow viewers to rewind
                        </label>
                      </div>

                      <div className="ui toggle checkbox">
                        <input
                          type="checkbox"
                          name="stream_ss5"
                          id="stream_ss5"
                          checked={isChecked5}
                          onChange={() => setIsChecked5(!isChecked5)}
                        />
                        <label htmlFor="stream_ss5">
                          End broadcast when stream ends
                        </label>
                      </div>

                      <div className="ui toggle checkbox _1457s2">
                        <input
                          type="checkbox"
                          name="stream_ss6"
                          id="stream_ss6"
                          checked={isChecked6}
                          onChange={() => setIsChecked6(!isChecked6)}
                        />
                        <label htmlFor="stream_ss6">
                          Publish as a test broadcast
                        </label>
                      </div>

                      <div className="ui toggle checkbox _1457s2">
                        <input
                          type="checkbox"
                          name="stream_ss7"
                          id="stream_ss7"
                          checked={isChecked7}
                          onChange={() => setIsChecked7(!isChecked7)}
                        />
                        <label htmlFor="stream_ss7">
                          Turn off live commentary
                        </label>
                      </div>
                      <button className="_145d1">Save changes & Go Live</button>
                    </div>
                  </div>
                )}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
