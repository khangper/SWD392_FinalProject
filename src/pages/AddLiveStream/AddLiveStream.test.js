import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import { screen, fireEvent } from "@testing-library/react";
import AddLiveStream from "./AddLiveStream";

// Helper function to mock clipboard API
const mockClipboard = () => {
  const clipboard = {
    writeText: jest.fn().mockResolvedValueOnce(),
  };
  Object.assign(navigator, { clipboard });
  return clipboard;
};

describe("AddLiveStream", () => {
  it("renders the AddLiveStream component", () => {
    const { container } = renderWithProviders(<AddLiveStream />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("changes tabs correctly", () => {
    renderWithProviders(<AddLiveStream />);

    // Click on the 'Setting' tab
    fireEvent.click(screen.getByText("Setting"));

    // Check that 'Setting' tab content is visible
    expect(screen.getByText("Live Streaming Setting")).toBeVisible();

    // Click on the 'Add Streaming' tab
    fireEvent.click(screen.getByText("Add Streaming"));

    // Check that 'Add Streaming' tab content is visible
    expect(
      screen.getByText("Connect your live stream to the Live API")
    ).toBeVisible();
  });

  it("checks if 'Go Live' button is rendered and clickable", () => {
    renderWithProviders(<AddLiveStream />);

    // Check if 'Go Live' button is rendered
    expect(screen.getByText("Go Live")).toBeInTheDocument();

    // Simulate click on 'Go Live' button
    fireEvent.click(screen.getByText("Go Live"));

    // Add more assertions here if there are any side effects when clicking 'Go Live'
  });

  it("copies server URL to clipboard", async () => {
    const clipboard = mockClipboard();
    renderWithProviders(<AddLiveStream />);

    // Simulate click on 'Copy' button for server URL
    fireEvent.click(screen.getAllByText("Copy")[0]);

    // Check if the clipboard.writeText method was called with the server URL
    expect(clipboard.writeText).toHaveBeenCalledWith(
      "rtmps://live-api-s.edututs+.com:443/rtmp/"
    );
  });

  it("copies stream key to clipboard", async () => {
    const clipboard = mockClipboard();
    renderWithProviders(<AddLiveStream />);

    // Simulate click on 'Copy' button for stream key
    fireEvent.click(screen.getAllByText("Copy")[1]);

    // Check if the clipboard.writeText method was called with the stream key
    expect(clipboard.writeText).toHaveBeenCalledWith(
      "592030151361629?s_bl=1&s_ps=1&s_sw=065&s_vt=api-s&a=AbzB7xYk7XdnKpBf"
    );
  });
});
