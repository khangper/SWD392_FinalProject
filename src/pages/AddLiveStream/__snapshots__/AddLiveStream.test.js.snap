// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLiveStream renders the AddLiveStream component 1`] = `
<div>
  <div
    class="sa4d25"
  >
    <div
      class="Live-container-fluid"
    >
      <div
        class="row justify-content-md-center"
      >
        <div
          class="col-lg-8 col-md-10"
        >
          <div
            class="section3125 stream_tabs"
          >
            <ul
              class="nav nav-tabs slive_tabs justify-content-center"
              id="myTab"
              role="tablist"
            >
              <li
                class="nav-item"
              >
                <div
                  aria-controls="add-streaming"
                  aria-selected="true"
                  class="nav-link active"
                  role="tab"
                >
                  Add Streaming
                </div>
              </li>
              <li
                class="nav-item"
              >
                <div
                  aria-controls="setting"
                  aria-selected="false"
                  class="nav-link "
                  role="tab"
                >
                  Setting
                </div>
              </li>
            </ul>
            <div
              class="tab-content"
              id="myTabContent"
            >
              <div
                class="tab-pane fade show active"
                id="add-streaming"
                role="tabpanel"
              >
                <div
                  class="add_stream_content"
                >
                  <h4
                    class="strm_title"
                  >
                    Connect your live stream to the Live API
                  </h4>
                  <div
                    class="sf475"
                  >
                    Use live-streaming software or a hardware encoder.
                     
                    <a
                      href="#"
                    >
                      Learn More
                    </a>
                  </div>
                  <div
                    class="stm_key"
                  >
                    Preview your broadcast with a stream key or paired encoder.
                  </div>
                  <div
                    class="live_form"
                  >
                    <div
                      class="group-form"
                    >
                      <label>
                        Server URL*
                      </label>
                      <div
                        class="input-group"
                      >
                        <input
                          class="_dlor1"
                          data-testid="server_url"
                          placeholder="Stream URL"
                          type="text"
                          value="rtmps://live-api-s.edututs+.com:443/rtmp/"
                        />
                        <button
                          class="_6tf7s red-button"
                          type="button"
                        >
                          Copy
                        </button>
                      </div>
                    </div>
                    <div
                      class="group-form"
                    >
                      <label>
                        Persistent stream key*
                      </label>
                      <div
                        class="input-group"
                      >
                        <input
                          class="_dlor1"
                          data-testid="stream_key"
                          placeholder="Stream Key"
                          type="text"
                          value="592030151361629?s_bl=1&s_ps=1&s_sw=065&s_vt=api-s&a=AbzB7xYk7XdnKpBf"
                        />
                        <button
                          class="_6tf7s red-button"
                          type="button"
                        >
                          Copy
                        </button>
                      </div>
                    </div>
                    <button
                      class="_145d1"
                    >
                      <i
                        class="uil uil-video"
                      />
                      Go Live
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
